system/core/database.php

```diff
@@ -560,6 +560,44 @@ function serveFile($filename, $root) {
 			die("Invalid filename: $filename");
 		}
 		header('Content-Type: ' . getMimeType($filename));
+	  $filesize = filesize($filename);
+		$shortlen =$filesize-1;
+
+		if(getMimeType($filename)==='audio/mpeg'){
+
+			foreach (getallheaders() as $name => $value) {
+			    if($name === 'Range'){
+						// bytes=0-50
+						$range = substr($value,6); // "bytes=start-end" -> "start-end"
+						$x = explode('-',$range); // "start-end" -> ["start","end"]
+						$start = intval($x[0]);
+						$end = intval($x[1]);
+						http_response_code(206);
+						$bytesize=$end-$start;
+						header( 'Content-Length: ' . (string)($bytesize));
+						header( 'Content-Range: bytes '.$start.'-'.$end.'/'.$filesize);
+						header("Pragma: public");
+				 		header("Expires: 0");
+					  header( 'Accept-Ranges: bytes');
```

Comment:
This should probably also be added when the full file is requested?

Reply:
Check